sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_common/math.py

```diff
@@ -0,0 +1,19 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import math
+from functools import reduce
+from typing import List
+
+
+def list_sum(lst: List[float]) -> float:
+    return reduce(lambda a, b: a+b, lst)
+
+
+def list_mean(lst: List[float]) -> float:
+    return list_sum(lst)/len(lst)
+
+
+def list_mean_nan_safe(lst: List[float]) -> float:
```

Comment:
why are we using two separate functions for mean? ie why not have all of our list mean operations be nan safe ?

Reply:
it's mainly about expectations-- if we know an array can contain NaN's, we always will choose to use a NaN-safe mean. However, if the data we are averaging is not expected to have NaN's, and we nanmean it anyways, we would not know that some of our data had erroneous NaN values and would report an incorrect mean. This is likely why numpy also has 2 mean functions (which is the pattern I was following here): `np.mean` (not NaN-safe) and `np.nanmean` (NaN-safe)